dulwich/porcelain.py

```diff
@@ -172,16 +172,29 @@ def open_repo_closing(path_or_repo):
 
 
 def path_to_tree_path(repopath, path):
-    """Convert a path to a path usable in e.g. an index.
+    """Convert a path to a path usable in an index, e.g. bytes and relative to
+    the repository root.
 
-    :param repo: Repository
-    :param path: A path
+    :param repopath: Repository path, absolute or relative to the cwd
+    :param path: A path, absolute or relative to either the cwd or repopath
```

Comment:
Allowing relative paths to either cwd or repopath makes the behaviour of this function unpredictable. This should just be cwd (to be compatible with c git).

Reply:
I agree, this was to stay compatible with check ignore command which seemed, according to tests, to necessitate this. Should i change tests or logic of that function then ?

Reply:
dulwich.porcelain.check_ignore should also behave similarly to "git check-ignore". It looks like this will indeed require changing the tests for check_ignore.

Reply:
done